node/external/blockAPI/shardBlock.go

```diff
@@ -83,6 +87,66 @@ func (sbp *shardAPIBlockProcessor) GetBlockByRound(round uint64, options api.Blo
 	return sbp.convertShardBlockBytesToAPIBlock(headerHash, blockBytes, options)
 }
 
+// GetAlteredAccountsForBlock will return the altered accounts for the desired shard block
+func (sbp *shardAPIBlockProcessor) GetAlteredAccountsForBlock(options api.GetAlteredAccountsForBlockOptions) (*common.AlteredAccountsForBlockAPIResponse, error) {
+	headerHash, blockBytes, err := sbp.getHashAndBlockBytesFromStorer(options.GetBlockParameters)
```

Comment:
Directly calling the public methods `GetBlockByNonce` / `GetBlockByHash` wouldn't have been helpful in this context?

Reply:
good idea. tried this, but it involves an additional DB query to compute the block status, which I don't need right now. If really desired, I will make the refactoring